src/Practitioner/Settings/index.js

```diff
@@ -1,215 +1,289 @@
-            </ul>
-        </div>
-    )
-}
-
 const NavItem = observer((props) => {
-    const { t } = useTranslation('translation');
-    const { practitionerUIStore } = useStores();
-    const classes = useStyles({
-        selected: practitionerUIStore.settingsTab === props.id,
-        isLogout: props.id === "logout"
-    });
-    const logout = useLogout();
-
-
-    const handleClick = (id) => {
-
-        if (id === "logout") {
-            logout();
-            return
-        }
-
-        if (!id) {
-            return
-        }
-
-        practitionerUIStore.settingsTab = props.id
-    }
-
-    return (<li className={classes.navItemContainer}>
-        <ButtonBase href={props.href} target="blank" onClick={() => { handleClick(props.id) }} className={classes.navItem}>
-            {props.icon}
-            <span>{props.text}</span>
-        </ButtonBase>
-    </li>)
-})
-
-const BodyRouter = observer((props) => {
-    const { practitionerUIStore } = useStores();
-    const { t } = useTranslation('translation');
-    const classes = useStyles();
-    return (
-        <div className={classes.body}>
-            <Typography className={classes.header} variant="h2">{t(`coordinator.settingsPage.${practitionerUIStore.settingsTab}`)}</Typography>
-            <div className={classes.bodyContent}>
-                {practitionerUIStore.settingsTab === "language" && <Box padding="2em 0"><Language
-                    selectedBackgroundColor={Colors.textDarkGray}
-                    selectedTextColor="white"
-                    defaultBackgroundColor={Colors.lightgray}
-                    defaultTextColor={Colors.textDarkGray}
-                    defaultBorderColor={Colors.gray}
-                    selectedBorderColor={Colors.gray}
-
-                />
-                </Box>}
-                {practitionerUIStore.settingsTab === "documents" && <Documents />}
-                {practitionerUIStore.settingsTab === "updatePassword" && <div className={classes.password}><PasswordReset /></div>}
-                {practitionerUIStore.settingsTab === "account" && <Profile />}
-                {practitionerUIStore.settingsTab === "patientInformation" && <div className={classes.patientInformation}>
-                    <Typography className={classes.topInfo} variant="body1">
-                        {t('coordinator.settingsPage.patientViewExplanation')}
-                    </Typography>
-                    <PatientInformation />
-                </div>}
-            </div>
-        </div>
+  const location = useLocation();
+  const { isMobile } = useWindowSize();
+
+  const classes = useStyles({
+    selected:
+      (props.to === 'documents' &&
+        location.pathname === '/settings' &&
+        !isMobile) ||
+      (props.to === location.pathname.split('/')[2] && props.to),
+    isLogout: props.isLogout,
+  });
 
-    )
-})
+  const logout = useLogout();
+
+  const LinkButton = (props) => (
```

Comment:
This should not be defined within another component, it will cause unnecessary re-renders

Reply:
Talkinga about the "LinkButton" component

Reply:
This makes sense. I'll look into how / why it would re-render unnecessarily just for future reference!

Reply:
Problem resolved.